### PR TITLE
GH-30: Implement ReturnItem Lambda Handler

### LINKED ISSUE #30: Implement ReturnItem Lambda Handler
Story Points: [1]

# How?

Look at Task Breakdown in [Design Doc](https://github.com/gp-seattle/rms-app/wiki/AWS-Amplify-Project:-Functions)

### PR DESCRIPTION
**Pull request by johnnyhchen**
resolves #30

### CODE REVIEW

File: amplify/backend/ts-code/__tests__/handlers/api/ReturnItem.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// make AWS Lambda function
+// name function: API Name in Upper Camel Case
+// ex: ReturnItem
+// pick Node JS runtime
+// use Hello Wolrd function template (we're overriding everything)
+//
+
+//in amplify/function
+// we deleted index.js because it was the function code
```
Comment (yauj) [inline amplify/backend/ts-code/__tests__/handlers/api/ReturnItem.test.ts:36]: Remove this

File: amplify/backend/ts-code/src/api/ReturnItem.ts [hunk 2]
```diff
@@ -46,19 +46,34 @@ export class ReturnItem {
...
+    public execute(input: ReturnItemInput): Promise<string> {
         return emitAPIMetrics(
             () => {
-                return Promise.all(scratch.ids.map((id: string) =>
-                    this.mainTable.changeBorrower(id, scratch.borrower, "return", scratch.notes)
-                )).then(() => `Successfully returned items '${scratch.ids.toString()}'.`)
+                return this.performAllFVAs(input)
+                    .then(() => {
+                        return Promise.all(input.ids.map((id: string) =>
+                        this.mainTable.changeBorrower(id, input.borrower, "return", input.notes)
```
Comment (yauj) [inline amplify/backend/ts-code/src/api/ReturnItem.ts:55]: Indent this i.e. press tab

**Review by yauj — COMMENTED** [review]
Revert package.json and package-lock.json changes.

**Review by DBarnes2 — COMMENTED** [review]
If I'm not mistaken some of the changes here are being reviewed in: https://github.com/gp-seattle/rms-app/pull/45, so once that gets approved we can rebase and see the new changes

**Comment by yauj** [comment]
> If I'm not mistaken some of the changes here are being reviewed in: #45, so once that gets approved we can rebase and see the new changes

Yes. I want to push Johnny's changes first and then mine. I think there would be some sort of conflict.

**Review by yauj — APPROVED** [review]
LGTM!

**Pull request merged**